src/util/newconfig/ConfigDescription.hpp

```diff
@@ -0,0 +1,125 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include "util/Assert.hpp"
+
+#include <algorithm>
+#include <array>
+#include <string_view>
+
+namespace util::config {
+
+/**
+ * @brief All the config description are stored and extracted from this class
+ *
+ * Represents all the possible config description
+ */
+struct ClioConfigDescription {
+public:
+    /** @brief Struct to represent a key-value pair*/
+    struct KV {
```

Comment:
Hmmm... i think i have seen this type somewhere. perhaps `std::pair`? 😃 
I don't mind rolling our own but you can also just do `using KV = std::pair<std::string_view, std::string_view>;`. The only downside would be the more generic names `first` and `second`.

Reply:
I think using pair in most cases decreases readability. So it's fine to define own structures.